datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1161,6 +1162,15 @@ impl SMJStream {
             let filter_columns = if chunk.buffered_batch_idx.is_some() {
                 if matches!(self.join_type, JoinType::Right) {
                     get_filter_column(&self.filter, &buffered_columns, &streamed_columns)
+                } else if matches!(self.join_type, JoinType::LeftSemi) {
```

Comment:
I wonder if this should also check for `JoinType::Left` (and the clause above also check for `JoinType::RightSemi` 🤔